rclpy/rclpy/node.py

```diff
@@ -349,6 +349,9 @@ def declare_parameter(
         This method, if successful, will result in any callback registered with
         :func:`add_on_set_parameters_callback` to be called.
 
+        The name and type in the given descriptor is ignored, and should be specified using
+        the name argument to this function and the default value's type instead.
```

Comment:
I think we need a tutorial about typed parameters without a default value and how to get things like an empty array of strings as a default value, which it's not well documented and it's a bit tricky.
But that's unrelated to this contribution.